include/init.h

```diff
@@ -25,6 +25,10 @@ extern "C" {
 #define _SYS_INIT_LEVEL_POST_KERNEL	2
 #define _SYS_INIT_LEVEL_APPLICATION	3
 
+#ifdef CONFIG_SMP
+#define _SYS_INIT_LEVEL_SMP		4
```

Comment:
I'm not sure I understand why we need a dedicated init level, wouldn't getting all this up and running at POST_KERNEL be sufficient?

Reply:
POST_KERNEL is done before `z_smp_init()` is called (in `bg_thread_main()`). For something that must be done after all cores are up and running, we need another init level.

Reply:
I see, wonder if we should just move z_smp_init() to z_cstart()?